pkg/apis/garden/validation/validation.go

```diff
@@ -1852,6 +1892,76 @@ func ValidateWorker(worker garden.Worker, fldPath *field.Path) field.ErrorList {
 	if len(worker.Taints) > 0 {
 		allErrs = append(allErrs, validateTaints(worker.Taints, fldPath.Child("taints"))...)
 	}
+	if worker.Kubelet != nil {
+		allErrs = append(allErrs, ValidateKubeletConfig(*worker.Kubelet, fldPath.Child("kubelet"))...)
+	}
+
+	return allErrs
+}
+
+// ValidateWorker validates the worker object.
+func ValidateKubeletConfig(kubeletConfig garden.KubeletConfig, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if kubeletConfig.MaxPods != nil {
+		allErrs = append(allErrs, apivalidation.ValidateNonnegativeField(int64(*kubeletConfig.MaxPods), fldPath.Child("maxPods"))...)
+		// PodCIDR is /24
+		if *kubeletConfig.MaxPods > 256 {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("maxPods"), *kubeletConfig.MaxPods, "the maximum number of pods on a node allowed by a kubelet cannot exceed 256 as the pod network CIDR is /24"))
+		}
+	}
+
+	if kubeletConfig.EvictionPressureTransitionPeriod != nil {
+		allErrs = append(allErrs, ValidatePositiveDuration(kubeletConfig.EvictionPressureTransitionPeriod, fldPath.Child("evictionPressureTransitionPeriod"))...)
+	}
+
+	if kubeletConfig.EvictionMaxPodGracePeriod != nil {
+		allErrs = append(allErrs, apivalidation.ValidateNonnegativeField(int64(*kubeletConfig.EvictionMaxPodGracePeriod), fldPath.Child("evictionMaxPodGracePeriod"))...)
+	}
+
+	if kubeletConfig.EvictionHard != nil {
+		fldPathEvictionHard := fldPath.Child("evictionHard")
+		allErrs = append(allErrs, ValidateResourceQuantityOrPercent(kubeletConfig.EvictionHard.MemoryAvailable, fldPathEvictionHard, "memoryAvailable")...)
```

Comment:
Create a separate function for this that takes the to-be-created struct with the five attributes and validates them. Then just call it multiple times.